app/js/editTests.js

```diff
@@ -0,0 +1,45 @@
+const queryString = window.location.search;
+const urlParams = new URLSearchParams(queryString);
+const testId = urlParams.get('test_id') || 1;
+
+let authorised = false;
+let user = getCookie('userEmail');
+
+getUser(user).then((user) => {
+    if (user.data.isAdmin === '1') {
+        authorised = true;
+    }
+});
+
+
+populateHandlebars('.tableBody', 'js/templates/questionTable.hbs', `question?test_id=${testId}`).then(() => {
+    document.querySelectorAll(".delete-question").forEach((btn) => {
+        btn.addEventListener('click', async (clickedBtn) => {
+            const { id } = clickedBtn.target.parentElement.dataset;
+            await deleteQuestion(id);
+            location.reload();
+        });
+    });
+
+    document.querySelectorAll(".edit-question").forEach((btn) => {
+        btn.addEventListener('click', async (clickedBtn) => {
+            const { id } = clickedBtn.target.parentElement.dataset;
+
+            window.location.href = `editQuestion.html#${id}`;
+        });
+    });
+});
+
+
+populateHandlebars('#filterTests', 'js/templates/testAllocatedFilter.hbs', 'test').then(() => {
+    document.querySelector('#filterTests').value = testId;
+    document.querySelector('#filterTests').addEventListener('click', (testFilter) => {
+        const {id} = testFilter.target.dataset;
+        window.location.href = `editTests.html?test_id=${id}`;
```

Comment:
why refresh the page and not just change the data? makes it particularly slow

Reply:
ah, you dont think it does change the page. yeh, the code under this does nothing, as its actually just refreshing the page with a new url. if you used hash locations for the url param it wouldnt do that